### PR TITLE
Match release optimization behavior of Xcode 7

### LINKED ISSUE #500: Optimize debug vs. release clang builds
We use the same optimization setting for debug and release builds.  One of those is probably wrong :)

### PR DESCRIPTION
**Pull request by advayDev1**
Fixes #500

### CODE REVIEW

File: src/main/groovy/com/github/j2objccontrib/j2objcgradle/NativeCompilation.groovy [hunk 1]
```diff
@@ -290,6 +290,8 @@ class NativeCompilation {
                     if (buildType == buildTypes.debug) {
                         objcCompiler.args '-g'
                         objcCompiler.args '-DDEBUG=1'
+                    } else {  // release
+                        objcCompiler.args '-Os'
```
Comment (brunobowden) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/NativeCompilation.groovy]: Can you add a comment referencing where this argument is described and also where it's explained why this is standard for release?

**Comment by brunobowden** [comment]
LGTM - please see comment about adding comments ;-)

File: src/main/groovy/com/github/j2objccontrib/j2objcgradle/NativeCompilation.groovy (hunk 1 above)
Comment (advayDev1) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/NativeCompilation.groovy]: done.  referenced clang source re: what it does.
however, there is no defined standard; i merely referenced we are matching Xcode 7's behavior.

**Comment by brunobowden** [comment]
So you verified the behaviour manually rather than seeing it documented somewhere?

**Comment by advayDev1** [comment]
What 'it' does is per llvm's source.  The fact that we should indeed do 'it' is verified by running Xcode.

**Comment by advayDev1** [comment]
Well found some documentation after the fact that validates my understanding:
https://developer.apple.com/library/ios/qa/qa1795/_index.html#//apple_ref/doc/uid/DTS40014195-CH1-COMPILER

will add as well

**Pull request merged**